lib/driver.js

```diff
@@ -657,11 +657,33 @@ class XCUITestDriver extends BaseDriver {
   }
 
   async startSim () {
-    await this.opts.device.run({
+    const runOpts = {
       scaleFactor: this.opts.scaleFactor,
       connectHardwareKeyboard: !!this.opts.connectHardwareKeyboard,
       isHeadless: !!this.opts.isHeadless,
-    });
+    };
+
+    // This is to workaround XCTest bug about changing Simulator
+    // orientation is not synchronized to the actual window orientation
+    const orientation = this.opts.orientation;
+    if (_.isString(orientation)) {
+      switch (orientation.toUpperCase()) {
```

Comment:
This transform should be on line 668. Seems conceptually very strange to run it on something that is a constant.

Reply:
corrected